sogs/routes/rooms.py

```diff
@@ -264,3 +269,39 @@ def message_unpin(room, msg_id):
 def message_unpin_all(room):
     room.unpin_all(g.user)
     return jsonify({})
+
+
+@rooms.post("/room/<Room:room>/file")
+@auth.user_required
+def upload_file(room):
+    """ upload a file to a room """
+
+    if not room.check_upload(g.user):
+        abort(http.FORBIDDEN)
+
+    filename = request.headers.get("X-Filename")
+    if not filename:
+        abort(http.BAD_REQUEST)
+    id = room.upload_file(request.data, g.user, filename=secure_filename(filename))
+    resp = make_response(jsonify({"id": id}))
+    resp.status_code = http.CREATED
+    return resp
+
+
+@rooms.get("/room/<Room:room>/file/<int:fileId>/<filename>")
+@auth.user_required
+def serve_file(room, fileId, filename):
+    """
+    serves a file uploaded to a room
+    """
+    room_file = room.get_file(fileId)
+    if not room_file:
+        abort(http.NOT_FOUND)
+
+    resp = make_response(send_from_directory(abspath(curdir), room_file.path, as_attachment=True))
+
+    resp.headers["Date"] = http_date(room_file.uploaded)
+    resp.headers["Expires"] = http_date(room_file.expiry)
```

Comment:
What happens here if `expiry` is None?

Reply:
will that ever happen?

Reply:
Yup: room images have a null expiry, as will attachments in pinned posts (though that isn't implemented yet).

Reply:
what should we send in this case, expiry is 0?

Reply:
I think just omit the Expiry header.

Reply:
From what I gather, there is no "never" value for Expiry headers, and a browser is supposed to treat anything that it fails to parse as 0 which it is supposed to mean "it expired a long time ago"